clients/java/zpe/src/main/java/com/yahoo/athenz/zpe/pkey/file/FilePublicKeyStore.java

```diff
@@ -89,18 +131,26 @@ void loadPublicKeys(ArrayList<PublicKeyEntry> publicKeys, Map<String, PublicKey>
     
     @Override
     public PublicKey getZtsKey(String keyId) {
-        if (keyId == null) {
-            return null;
-        }
-        return ztsPublicKeyMap.get(keyId);
+        return getPublicKey(keyId, ztsPublicKeyMap);
     }
 
     @Override
     public PublicKey getZmsKey(String keyId) {
+        return getPublicKey(keyId, zmsPublicKeyMap);
+    }
+
+    private PublicKey getPublicKey(String keyId, Map<String, PublicKey> ztsPublicKeyMap) {
         if (keyId == null) {
             return null;
         }
-        return zmsPublicKeyMap.get(keyId);
+        PublicKey publicKey = ztsPublicKeyMap.get(keyId);
+        if (publicKey == null) {
+            // reload athenz jwks from disk and try again
+            LOG.debug("key id: {} does not exist in public keys map, reload athenz jwks from disk", keyId);
+            initAthenzJWKConfig();
```

Comment:
let's also have similar logic not read the file every time - e.g. once every 30 mins